lua/terrortown/menus/gamemode/administration/playermodels.lua

```diff
@@ -63,43 +63,46 @@ function CLGAMEMODESUBMENU:Populate(parent)
 			end
 		}, base)
 
-		playermodels.IsSelectedModel(name, function(value)
-			boxCache[name]:SetModelSelected(value, false)
-		end)
+		-- Add additional checks to ensure boxCache[name] exists
+		if boxCache[name] then
```

Comment:
This check should be unnecessary. We created boxcache{name] a few lines before.  You rather need to check it in the callback functions, but I believe that they are synced anyways.

Reply:
Just checked it, playermodels are not necessarily synced. So move the boxCache[name] check into playermodels.IsSelectedModel and playermodels.IsHattableModel.

Reply:
Is that so?

Reply:
Please dont resolve comments yourself.
And this is not what I meant. you dont need to check boxCache[name] before adding the anonymous function, you need to check it inside of the anonymous function.